### PR TITLE
Ensures area contents respects world.maxx/y expanding

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

It never like, happens, and if it ever did we'd yell at the guy who did it
But if we're gonna pretend we support expanding in the maploader, we should support it structurally too

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -360,12 +360,34 @@ GLOBAL_VAR(restart_counter)
...
 
+// If this is called as a part of maploading you cannot call it on the newly loaded map zs, because those get handled later on in the pipeline
+/world/proc/increaseMaxX(new_maxx, max_zs_to_load = maxz)
+	if(new_maxx <= maxx)
+		return
+	var/old_max = world.maxx
+	maxx = new_maxx
+	if(!max_zs_to_load)
+		return
+	var/area/global_area = GLOB.areas_by_type[world.area] // We're guarenteed to be touching the global area, so we'll just do this
```
Comment (Mothblocks) [inline code/game/world.dm:371]: `guaranteed`

Comment (LemonInTheDark) [reply code/game/world.dm:371]: AW FUCK

Comment (LemonInTheDark) [reply code/game/world.dm:371]:
```suggestion
	var/area/global_area = GLOB.areas_by_type[world.area] // We're guaranteed to be touching the global area, so we'll just do this
```
you know the worst part I noticed this earlier today in something else I wrote

Comment (LemonInTheDark) [reply code/game/world.dm:371]:
```suggestion
	var/area/global_area = GLOB.areas_by_type[world.area] // We're guaranteed to be touching the global area, so we'll just do this
```
you know the worst part I noticed this earlier today in something else I wrote

Comment (LemonInTheDark) [reply code/game/world.dm:371]:
```suggestion
	var/area/global_area = GLOB.areas_by_type[world.area] // We're guaranteed to be touching the global area, so we'll just do this
```
you know the worst part I noticed this earlier today in something else I wrote

**Pull request merged**